x/ccv/provider/migrations/v8/migrations.go

```diff
@@ -0,0 +1,113 @@
+package v8
+
+import (
+	"encoding/binary"
+	"time"
+
+	storetypes "cosmossdk.io/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	providerkeeper "github.com/cosmos/interchain-security/v5/x/ccv/provider/keeper"
+	providertypes "github.com/cosmos/interchain-security/v5/x/ccv/provider/types"
+)
+
+const (
+	LegacyUnbondingOpBytePrefix                = byte(11)
+	LegacyConsumerAddrsToPruneBytePrefix       = byte(25)
+	LegacyMaturedUnbondingOpsByteKey           = byte(1)
+	LegacyUnbondingOpIndexBytePrefix           = byte(12)
+	LegacyInitTimeoutTimestampBytePrefix       = byte(8)
+	LegacyVscSendTimestampBytePrefix           = byte(18)
+	LegacyVSCMaturedHandledThisBlockBytePrefix = byte(28)
+)
+
+// CompleteUnbondingOps completes all unbonding operations.
+// Note that it must be executed before CleanupState.
+func CompleteUnbondingOps(ctx sdk.Context, store storetypes.KVStore, pk providerkeeper.Keeper) {
+	iterator := storetypes.KVStorePrefixIterator(store, []byte{LegacyUnbondingOpBytePrefix})
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		id := binary.BigEndian.Uint64(iterator.Key()[1:])
+		if err := pk.UnbondingCanComplete(ctx, id); err != nil {
+			pk.Logger(ctx).Error("UnbondingCanComplete failed", "unbondingID", id, "error", err.Error())
+		}
+	}
+}
+
+// MigrateConsumerAddrsToPrune migrates the ConsumerAddrsToPrune index to ConsumerAddrsToPruneV2.
+// Note: This migration must be done before removing the VscSendTimestamp index
+func MigrateConsumerAddrsToPrune(ctx sdk.Context, store storetypes.KVStore, pk providerkeeper.Keeper) {
```

Comment:
Since this is a relatively complex migration, should there be a test for it? e.g. compare v7/migrations_test.go

Reply:
Probably, but it's not easy to make one as we need a bunch of functions that were removed :(

Reply:
See https://github.com/cosmos/interchain-security/pull/2098/commits/eafb76f6c6a80907acbf817a417e9805b15d430c